dags/als-kp-dataset-dag.py

```diff
@@ -219,49 +217,116 @@ def transform_data(data: Dict[str, Any], **context) -> List[Dict[str, Any]]:
-        """Create Synapse datasets and collection.
+    def find_duplicated_datasets(
+        transformed_items: List[Dict[str, Any]], **context
+    ) -> List[Dict[str, Any]]:
+        """Flag duplicated new C-PATH datasets.
 
         This task:
-        1. Creates a new Dataset Collection with predefined columns
-        2. For each transformed item:
-           - Creates a new Dataset with the item's title and description
-           - Adds the Dataset to the collection
-        3. Stores the collection in Synapse
+        1. Retrieves the current C-PATH datasets.
+        2. Identifies and flags new datasets that are duplicates based on their 'title'.
 
-        Arguments:
-            transformed_items: List of transformed and validated items
-            **context: Airflow task context containing DAG parameters
+        Args:
+            transformed_items (List[Dict[str, Any]]):
+                A list of transformed and validated items from the previous task.
+
+        Raises:
+            ValueError: If any 'sameAs' or 'title' values are missing or empty after transformation.
 
         Returns:
-            str: The Synapse ID of the created and stored dataset collection
+            List[Dict[str, Any]]: A list of duplicate dataset items if there are any.
         """
         syn_hook = SynapseHook(context["params"]["synapse_conn_id"])
         synapse_client = syn_hook.client
 
-        columns = [
-            Column(name="id", column_type=ColumnType.ENTITYID),
-            Column(name="title", column_type=ColumnType.STRING, maximum_size=200),
-            Column(name="creator", column_type=ColumnType.STRING, maximum_size=100),
-            Column(name="keywords", column_type=ColumnType.STRING, maximum_size=250),
-            Column(name="subject", column_type=ColumnType.STRING, maximum_size=100),
-            Column(name="collection", column_type=ColumnType.STRING, maximum_size=100),
-            Column(name="publisher", column_type=ColumnType.STRING, maximum_size=100),
-            Column(name="species", column_type=ColumnType.STRING, maximum_size=100),
-            Column(name="sameAs", column_type=ColumnType.STRING, maximum_size=100),
-            Column(name="source", column_type=ColumnType.STRING, maximum_size=100),
-            Column(name="url", column_type=ColumnType.STRING, maximum_size=100),
-        ]
-
+        # Get current datasets
         dataset_collection = DatasetCollection(
-            name=context["params"]["collection_name"],
-            description=context["params"]["collection_description"],
-            parent_id=context["params"]["project_id"],
-            include_default_columns=True,
-            columns=columns,
+            id=context["params"]["collection_id"]
+        ).get(synapse_client=synapse_client)
+        current_data = dataset_collection.query(
+            query=f"SELECT * from {dataset_collection.id} where publisher='Critical Path Institute'",
```

Comment:
Is this `Critical Path Institute` always going to be the publisher or should this be programmatically fed?

Reply:
Here, we are trying to retrieve all the C-Path datasets, and yes, all the C-Path datasets have publisher = Critical Path Institute